tool/infocert.c

```diff
@@ -0,0 +1,119 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+#include "infocert.h"
+#include <xtt/crypto_types.h>
+#include <xtt/certificates.h>
+#include <xtt/util/asn1.h>
+#include <xtt/util/root.h>
+#include <xtt/util/util_errors.h>
+#include <xtt/util/file_io.h>
+#include <string.h>
+#include <stdio.h>
+
+struct xtt_server_certificate_raw_type;
+
+static
+void server_cert_out(unsigned char* certificate){
+    unsigned char* server_id = xtt_server_certificate_access_id((struct xtt_server_certificate_raw_type*) certificate);
+    printf("Server ID: ");
+    for(unsigned int i = 0; i < sizeof(xtt_identity_type); i++){
+        if(server_id[i]<0x10)
```

Comment:
The `printf` function can already print the leading zero for `0-9` for you.

So replace the `if < 0x10; printf("0x"); else printf("x")` constructs with

```
printf("%02x\n", buffer[i]);
```

This says print the byte in hex form (x) using two columns (2) and left padding with `0`s (0) if the text is less than 2 columns.